src/API/Site/Controllers/MerchantCenter/PriceBenchmarksController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query\MerchantPriceSuggestionsQuery;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\MerchantPriceBenchmarks;
+use Automattic\WooCommerce\GoogleListingsAndAds\Internal\ContainerAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Internal\Interfaces\ContainerAwareInterface;
+use Exception;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PriceBenchmarksController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class PriceBenchmarksController extends BaseController implements ContainerAwareInterface {
+
+	use ContainerAwareTrait;
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'mc/price-benchmarks',
+			[
+				[
+					'methods'  => TransportMethods::READABLE,
+					'callback' => $this->get_price_benchmarks_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Maps query arguments from the REST request.
+	 *
+	 * @param Request $request REST Request.
+	 * @return array
+	 */
+	protected function prepare_query_arguments( Request $request ): array {
+		$args = wp_parse_args(
+			array_intersect_key(
+				$request->get_query_params(),
+				$this->get_collection_params()
+			),
+			$request->get_default_params()
+		);
+
+		return $args;
+	}
+
+	/**
+	 * Get the callback function for the price benchmarks request.
+	 *
+	 * @return callable
+	 */
+	protected function get_price_benchmarks_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				/** @var MerchantPriceBenchmarks $merchant */
+				$merchant = $this->container->get( MerchantPriceBenchmarks::class );
+
+				$benchmark_data = $merchant->get_benchmark_data( 'products', $this->prepare_query_arguments( $request ) );
+				$benchmark_data = $this->prepare_item_for_response( $benchmark_data, $request );
+
+				$get_price_insights_product_view = $merchant->get_price_insights_product_view( 'products', $this->prepare_query_arguments( $request ) );
+				$get_price_insights_product_view = $this->prepare_item_for_response( $get_price_insights_product_view, $request );
+
+				return new Response(
```

Comment:
Once we have working data coming back from these responses, we'll need to stitch together data from both queries by matching the product IDs in order to return the expected response shape

Reply:
Yes, since we’re not getting any data for now, I’ve posted both responses in an object.

What do I need to set up to retrieve data from the merchant account, and in what format should we return the object data?

Reply:
@joemcgill I tried adding the unit tests, but I couldn’t get it working maybe I missed something while mocking the service or the controller. I found it a bit tricky to set up.  

Could you please share some guidance or pointers so I can work on it more effectively?

Reply:
Try to add unit test in https://github.com/woocommerce/google-listings-and-ads/pull/2838/commits/49f7823455d72be36a4bcf988e6795dd2e60369b

Reply:
You got pretty close! I've updated the test class in [c7e917a](https://github.com/woocommerce/google-listings-and-ads/pull/2838/commits/c7e917af0ae6583ab41d5b6a0ef640ca922277d5) to fix the setUp method and to get a better idea of what the expected shape should be.

The `PriceBenchmarksController` makes use of the `MerchantPriceBenchmarks` service class by getting it from the Container, so we needed to set up a mock Container and share a local `MerchantPriceBenchmarks` class that we then mock return values for.

Reply:
@joemcgill In https://github.com/woocommerce/google-listings-and-ads/pull/2838/commits/4833d054b89d9ec06ac932055cc0824dcc4a3f75 i shape the response but i don't get why the unit tests going failed for `$this->product_helper->get_wc_product_id` as when i run in browser with mock data it return proper ID.